staging/src/github.com/openshift/oc/pkg/cli/version/version.go

```diff
@@ -1,25 +1,199 @@
 package version
 
 import (
+	"encoding/json"
 	"fmt"
-	"io"
 
 	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
 
-	"k8s.io/apimachinery/pkg/version"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/tools/clientcmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	kversion "k8s.io/kubernetes/pkg/kubectl/cmd/version"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+	"k8s.io/kubernetes/pkg/kubectl/util/templates"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/oc/pkg/version"
 )
 
-// NewCmdVersion provides a shim around version for
-// non-client packages that require version information
-func NewCmdVersion(fullName string, versionInfo version.Info, out io.Writer) *cobra.Command {
+type Version struct {
+	kversion.Version
+	OpenShiftVersion string `json:"openshiftVersion,omitempty" yaml:"openshiftVersion,omitempty"`
```

Comment:
is the yaml tag needed? The yaml marshaller uses the json tag anyway, does't it?

Reply:
not needed.
One thing I'm not sure about, by using kversion.Version, yaml/json output doesn't match string output, bc it shows the kversion.Version types.  Should I keep the output `Server Version | Client Version | OpenShift Server Version`?  Then we'll keep very closely aligned w/ upstream and what we had in 3.x.    
```console
$ oc version -o yaml
clientVersion:
  buildDate: "2019-07-16T17:45:06Z"
  compiler: gc
  gitCommit: 44c27aaf67
  gitTreeState: dirty
  gitVersion: v4.1.0+44c27aa-1018-dirty
  goVersion: go1.12.5
  major: "4"
  minor: 1+
  platform: linux/amd64
openshiftVersion: 4.2.0-0.ci-2019-07-14-223252
serverVersion:
  buildDate: "2019-07-13T05:49:43Z"
  compiler: gc
  gitCommit: 65ef059
  gitTreeState: clean
  gitVersion: v1.14.0+65ef059
  goVersion: go1.12.5
  major: "1"
  minor: 14+
  platform: linux/amd64
```

Reply:
Is that the struct upstream uses to get the output?

Reply:
yes, json/yaml match the string output now.